src/GafferImage/Ramp.cpp

```diff
@@ -0,0 +1,280 @@
+{
+	ImageNode::hashDataWindow( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+Imath::Box2i Ramp::computeDataWindow( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue().getDisplayWindow();
+}
+
+IECore::ConstCompoundDataPtr Ramp::computeMetadata( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return outPlug()->metadataPlug()->defaultValue();
+}
+
+void Ramp::hashChannelNames( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelNames( output, context, h );
+	layerPlug()->hash( h );
+}
+
+IECore::ConstStringVectorDataPtr Ramp::computeChannelNames( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	std::string channelNamePrefix = layerPlug()->getValue();
+	if( !channelNamePrefix.empty() )
+	{
+		channelNamePrefix += ".";
+	}
+
+	StringVectorDataPtr resultData = new StringVectorData();
+	vector<string> &result = resultData->writable();
+
+	result.push_back( channelNamePrefix + "R" );
+	result.push_back( channelNamePrefix + "G" );
+	result.push_back( channelNamePrefix + "B" );
+	result.push_back( channelNamePrefix + "A" );
+
+	return resultData;
+}
+
+void Ramp::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelData( output, context, h );
+
+	V2i tileOrigin = context->get<V2i>( ImagePlug::tileOriginContextName );
+	h.append( tileOrigin );
+
+	string channelName = context->get<std::string>( ImagePlug::channelNameContextName );
+	h.append( channelName );
+
+	rampPlug()->hash( h );
+	transformPlug()->hash( h );
+
+	h.append( positionAPlug()->getValue() );
+	h.append( positionBPlug()->getValue() );
+}
+
+IECore::ConstFloatVectorDataPtr Ramp::computeChannelData( const std::string &channelName, const Imath::V2i &tileOrigin, const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	const int channelIndex = ImageAlgo::colorIndex( context->get<std::string>( ImagePlug::channelNameContextName ) );
+
+	const SplinefColor4f ramp = rampPlug()->getValue().spline();
+
+	const M33f transform = transformPlug()->matrix();
+	const V2f positionA = positionAPlug()->getValue();
+	const V2f positionB = positionBPlug()->getValue();
+
+	V3f positionA3f( positionA.x, positionA.y, 0 );
+	V3f positionB3f( positionB.x, positionB.y, 0 );
+
+	const LineSegment3f line( positionA3f, positionB3f );
+
+	FloatVectorDataPtr resultData = new FloatVectorData;
+	vector<float> &result = resultData->writable();
+	result.reserve( ImagePlug::tileSize() * ImagePlug::tileSize() );
+	
+	V3f closest;
+	float pos;
+	Color4f color;
+	for( int y = 0; y < ImagePlug::tileSize(); ++y )
+	{
+		for( int x = 0; x < ImagePlug::tileSize(); ++x )
+		{
+
+			// screen space pixel coordinates
+			V2f p( tileOrigin.x + x + .5f, tileOrigin.y + y + .5f);
+			p *= transform.inverse();
```

Comment:
If you [lift the inverse outside of this loop](https://github.com/johnhaddon/gaffer/commit/90cd1de67ce14d67b551d5a248fa3635cf8ae3cc) , there's a small performance boost to be had.